sormas-api/src/main/java/de/symeda/sormas/api/epidata/EpiDataHelper.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2018 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *******************************************************************************/
+package de.symeda.sormas.api.epidata;
+
+import de.symeda.sormas.api.utils.YesNoUnknown;
+
+public final class EpiDataHelper {
+
+	public static YesNoUnknown getKindOfExposure (EpiDataDto dto) {
```

Comment:
As far as I understand it, this method is used to determine whether the "kindOfExposure" fields are shown in the UI; please rename it to something like "hadExposureToAnimals"; also, if it's used for this purpose, it's probably better to make this method return a boolean. If I'm wrong, please explain what this method is supposed to do

Reply:
I've put some thought into your suggestion. I think we should keep the name and return type as is. This method is used to tell whether any `kindOfExposure` field has been ticked. 
If none is ticked, we set the `kindOfExposure` field to UNKNOWN and hide all related fields. 
If any field is ticked, we set it to YES and show all related fields.

With your suggestion, `hadExposureToAnimals` will only check whether any exposure field is set to `YES`. 
If so, it will show the `kindOfExposure` field, else it'll hide it.
In this way, when the user sets a field to `NO` or `UNKNOWN`, the system will still hide all the fields although some fields have been ticked (albeit not the `YES` option).

Maybe we can give this method a better name to describe what it does. Like `isAnyKindOfExposureFieldTicked` ;)

Reply:
Your suggestion makes sense, but makes me wonder whether this whole logic makes sense. Because, in theory, when all the individual `kindOfExposure` fields are set to `NO`, the custom `Kind of exposure` field should be set to `NO` as well; however, we typically hide depending fields when a parent field is set to `NO` and this would break that pattern.

Additionally, it's confusing that the caption of the generated field is `Kind of exposure`, but it's a `YesNoUnknown` field; As far as I understand it, the question whether the patient was exposed to animals in the first place is answered with the fields `dogs`, `cats`, `canidae` etc., right? Maybe it's a better idea to make the visibility status of the individual `kindOfExposure` fields dependent on whether one of these animal fields is set to `YES` and get rid of the generated `kindOfExposure` field completely? What do you think?

Reply:
The whole idea of visibility toggling is to avoid showing certain options unnecessarily. 
I'm thinking if we get rid of the kindOfExposure logic, we'll have to always show the dependent fields irrespective of their values. This is true, at least, for the create and edit views.
If it is okay to always show all the exposure fields, then we can take this route.

In another perspective, your initial comment is making a comeback. Ideally, "Kind of exposure" is a section that can be expanded or collapsed. It is initially (on loading of the view) collapsed to make room for other fields. If the user wants to fill those fields, he can expand it and tick appropriate fields.
With this logic, dependent fields do not have to clearOnHide because the user is allowed to collapse and expand the section at will. However, on initial loading of the view, we expand it if any field in the section is filled.

Reply:
That sounds like a cool piece of logic! I wouldn't use a `YesNoUnknown` field for the expand/collapse part though, but probably just a label with an arrow to the left. This UI element needs to be created first, which is an effort we probably can't make right now because the top priority is to get the new features into the system by Friday. However, I'll make sure to keep it in mind!

In the meantime, I suggest to get rid of the generated `kindOfExposure` field and make the visibility of the specialised `kindOfExposure` fields dependent on the various fields that determine contact with animals (because it doesn't make sense to specify the kind of exposure if no contact with animals took place; this way, we make sure to always hide these fields if they're not relevant at all). This also means that you can keep the method in the `EpiDataHelper`, but only have to change the fields that are taken into account. The new name you've already given it also makes sense in this context.

Are you okay with that approach for now?

Reply:
Yep. Sounds good.